src/player.cpp

```diff
@@ -10102,6 +10102,20 @@ int player::get_wind_resistance(body_part bp) const
     return totalCoverage;
 }
 
+int player::get_weight() const
+{
+    int ret = 0;
+    int wornWeight = 0;
+    for (auto itm : worn) {
```

Comment:
Could you possibly verify this addresses both concerns? Not familiar with std::accumulate or lambda functions:
```C++
    wornWeight = std::accumulate(worn.begin(),
                                 worn.end(),
                                 0,
                                 [](int sum, const item& itm) { return sum + itm.weight(); }
                                 );
```